src/core/validate-graph.ts

```diff
@@ -60,6 +60,18 @@ export function validatePackageURL(pkg: types.PkgInfo): void {
         );
         break;
 
+      // The PURL spec for Linux distros does not include the source in the name,
+      // but some of our purl generators erroneously include it. This is why we relax
```

Comment:
I'm not sure what you mean with this "some of our purl generators include it". 

We don't include the source in the (purl) name, only in the depgraph's package name.